spec/requests/topics_controller_spec.rb

```diff
@@ -1168,6 +1181,219 @@ def extract_post_stream
 
       expect(response.headers['X-Robots-Tag']).to eq(nil)
     end
+
+    describe "themes" do
+      let(:theme) { Theme.create!(user_id: -1, name: 'bob', user_selectable: true) }
+      let(:theme2) { Theme.create!(user_id: -1, name: 'bobbob', user_selectable: true) }
+
+      before do
+        sign_in(user)
+      end
+
+      it "selects the theme the user has selected" do
+        user.user_option.update_columns(theme_key: theme.key)
+
+        get "/t/666"
+        expect(controller.theme_key).to eq(theme.key)
```

Comment:
Hmm is `controller` made available somewhere in the test?

Reply:
From my observation, it seems the `controller` (along with `session`, `request` etc) objects become available after you make the request. So in this case, `controller` is an instance of TopicController that handled the request above.